src/ORM/Table.php

```diff
@@ -1728,7 +1728,8 @@ protected function _update($entity, $data)
         }
 
         if (!$entity->has($primaryColumns)) {
-            $message = 'All primary key value(s) are needed for updating';
+            $message = 'All primary key value(s) are needed for updating, ';
+            $message .= get_class($entity) . ' is missing ' . implode(', ', $primaryColumns);
```

Comment:
"is missing" might not be fully true tough for composite keys, maybe it would better if it said something like "requires", or checked which columns exactly are missing?